core/sync/sync.go

```diff
@@ -98,8 +100,36 @@ func RunLocalClockSync(log *zap.Logger, lclk timebase.LocalClock) {
 			if float64(timemath.Abs(corr)) > maxCorr {
 				corr = time.Duration(float64(timemath.Sign(corr)) * maxCorr)
 			}
-			// lclk.Adjust(corr, refClkInterval, 0)
-			pll.Do(corr, 1000.0 /* weight */)
+
+			theilSen.AddSample(corr)
+			offsetNs := theilSen.OffsetNs()
+			frequencyPPB := offsetNs/float64(refClkInterval.Nanoseconds())*1e9 + theilSen.baseFreq
+
+			log.Debug("Prediction from Theil-Sen: ",
+				zap.Float64("offset", offsetNs),
+				zap.Float64("freq (PPB)", frequencyPPB),
+			)
+
+			correction, interval, startFreq := pll.AddSampleAndGetData(corr, 1000.0 /* weight */)
+			pllFreqPPB := (startFreq + (correction / interval)) * 1e9
+
+			log.Debug("Prediction from PLL: ",
+				zap.Float64("correction", correction),
+				zap.Float64("interval", interval),
+				zap.Float64("startFreq", startFreq),
+				zap.Float64("freq (PPB)", pllFreqPPB),
+			)
+
+			switch algo {
+			case ClockAlgoPLL:
+				if interval > 0.0 {
+					lclk.Adjust(timemath.Duration(correction), timemath.Duration(interval), startFreq)
+				}
+			case ClockAlgoTS:
+				if math.Abs(frequencyPPB) > 0 {
```

Comment:
I once wanted to have a higher cutoff for adjustments because I thought this might make the clock more stable, but if that is done it should probably rather be done at an algorithm level given that a cutoff value might differ between algorithms.